src/core/conn.h

```diff
@@ -43,6 +40,22 @@ class CConnectionPoint {
     bool operator!=(const CConnectionPoint & paRight) const {
       return !(*this == paRight);
     }
+    
+    CFunctionBlock &getFB() {
+      return *mFB;
+    }
+    
+    CFunctionBlock &getFB() const {
```

Comment:
shouldn't this return a const&?

Reply:
but if the non-const method is needed, the above one will be used, right?

Reply:
Only if the SConnectionPoint is non const. We have one use case where the SConnectionPoint should not change but we need to update the contained FB. So SConnectionPoint is const but the FB itself not. How to implement this?

Reply:
For that particular case, I think this solution is the only one. But I'd change the caller actually to iterate over non-const connections. The caller looks like:

```
      for(const auto& it : mDestinationIds){
         it.getFB().connectDI(it.getPortId(), this);
       }
```

which to me seems counter-intuitive that `const` is specified to perform a non-const operation. The idea of  a `const CConnectionPoint` doesn't make much sense otherwise.

Reply:
Fixed this. For other use cases I will see where this whole thing leads us.